e2e/test/TwinE2ETests.cs

```diff
@@ -234,10 +284,21 @@ private async Task Twin_DeviceSetsReportedPropertyAndGetsItBackSingleDeviceAsync
             await Twin_DeviceSetsReportedPropertyAndGetsItBackAsync(deviceClient).ConfigureAwait(false);
         }
 
-        public static async Task Twin_DeviceSetsReportedPropertyAndGetsItBackAsync(DeviceClient deviceClient)
+        private async Task Twin_DeviceSetsReportedPropertyArrayAndGetsItBackSingleDeviceAsync(Client.TransportType transport)
+        {
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(_devicePrefix).ConfigureAwait(false);
+            using var deviceClient = DeviceClient.CreateFromConnectionString(testDevice.ConnectionString, transport);
+
+            await Twin_DeviceSetsReportedPropertyAndGetsItBackAsync(deviceClient, JArray.Parse("[1, 2, 3]")).ConfigureAwait(false);
+        }
+
+        public static async Task Twin_DeviceSetsReportedPropertyAndGetsItBackAsync(DeviceClient deviceClient, object propValue = null)
         {
             var propName = Guid.NewGuid().ToString();
-            var propValue = Guid.NewGuid().ToString();
+            if (propValue == null)
+            {
+                propValue = Guid.NewGuid().ToString();
```

Comment:
Why is this necessary? Can't you ignore if there is no propvalue?

Reply:
The tests that call this method provide the value that should be tested (a twin array, for example), or they provide nothing if they are fine with the GUID

Reply:
So would SDK work even without GUID and propvalue = null

Reply:
I'm not sure I follow. The SDK may or may not allow null values in the value field, but this PR has nothing to do with that

Reply:
This is just a default property value to use for a test of reported/desired properties, and this PR doesn't change anything about that. It just allows for the test to specify a different value to use

Reply:
Previously the tests were setting guid strings for both property name and property value, in this test block, and that was being used for updating twin properties. 
What you are trying to achieve now is, for new tests you are setting the `propValue` to be an array (which you are passing into the test method), while existing tests do not have `propValue` passed in, and hence will be using the above block to set a guid string.
The way this test block is structured is potentially causing the confusion. You should pass in `propValue` into the test method for string case as well, to avoid this confusion.

Reply:
I can remove the default value from this method parameter if that clears it up, sure